src/OrchardCore.Modules/OrchardCore.Media/Services/ChunkFileUploadService.cs

```diff
@@ -0,0 +1,242 @@
+
+            return await completedAsync(new[] { chunkedFormFile });
+        }
+        finally
+        {
+            DeleteTemporaryFile(uploadId, formFile);
+        }
+    }
+
+    public void PurgeTempDirectory()
+    {
+        var tempFolderPath = GetTempFolderPath();
+        if (_options.Value.TemporaryFileLifetime <= TimeSpan.Zero ||
+            !Directory.Exists(tempFolderPath))
+        {
+            return;
+        }
+
+        var tempFiles = Directory.GetFiles(tempFolderPath, $"{_shellSettings.TenantId}_*")
+            .Select(filePath => new FileInfo(filePath))
+            .Where(fileInfo => fileInfo.LastWriteTimeUtc + _options.Value.TemporaryFileLifetime < _clock.UtcNow)
+            .ToArray();
+
+        foreach (var tempFile in tempFiles)
+        {
+            if (!DeleteTemporaryFile(tempFile.FullName))
+            {
+                break;
+            }
+        }
+    }
+
+    private bool DeleteTemporaryFile(Guid uploadId, IFormFile formFile) =>
+        DeleteTemporaryFile(GetTempFilePath(uploadId, formFile));
+
+    private bool DeleteTemporaryFile(string tempFilePath)
+    {
+        try
+        {
+            File.Delete(tempFilePath);
+        }
+        catch (Exception exception) when (exception.IsFileSharingViolation())
+        {
+            _logger.LogError(
+                exception,
+                "An error occurred while deleting the temporary file '{TempFilePath}'.",
+                tempFilePath);
+
+            return false;
+        }
+
+        return true;
+    }
+
+    private Stream GetOrCreateTemporaryFile(Guid uploadId, IFormFile formFile, long size)
+    {
+        var siteTempFolderPath = GetTempFolderPath();
+
+        if (!Directory.Exists(siteTempFolderPath))
+        {
+            Directory.CreateDirectory(siteTempFolderPath);
+        }
+
+        var tempFilePath = GetTempFilePath(uploadId, formFile);
+
+        return File.Exists(tempFilePath) switch
+        {
+            true => File.Open(tempFilePath, FileMode.Open, FileAccess.ReadWrite),
+            false => CreateTemporaryFile(tempFilePath, size),
+        };
+    }
+
+    private ChunkedFormFile GetTemporaryFileForRead(Guid uploadId, IFormFile formFile)
+    {
+        var tempFilePath = GetTempFilePath(uploadId, formFile);
+
+        return new(File.OpenRead(tempFilePath))
+        {
+            ContentType = formFile.ContentType,
+            ContentDisposition = formFile.ContentDisposition,
+            Headers = formFile.Headers,
+            Name = formFile.Name,
+            FileName = formFile.FileName,
+        };
+    }
+
+    private static string GetTempFolderPath() =>
+        Path.Combine(Path.GetTempPath(), TempFolderPrefix);
+
+    private string GetTempFilePath(Guid uploadId, IFormFile formFile) =>
+        Path.Combine(
+            GetTempFolderPath(),
+            $"{_shellSettings.TenantId}_{CalculateHash(uploadId.ToString(), formFile.FileName, formFile.Name)}");
```

Comment:
Maybe init once the prefix `_tempFilePrefix = $"{_shellSettings.TenantId}_"` in the ctor, but as you want.